src/Controls/tests/UITests/Tests/Issues/Issue18961.cs

```diff
@@ -0,0 +1,56 @@
+﻿using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.AppiumTests.Issues
+{
+	public class Issue18961 : _IssuesUITest
+	{
+		const string ScrollButton = "ScrollButton";
+		const string LastEntry = "TestEntry20";
+
+		public Issue18961(TestDevice device) : base(device) { }
+
+		public override string Issue => "Modal Page margin correct after Keyboard opens";
+
+		[Test]
+		public async Task ModalPageMarginCorrectAfterKeyboardOpens()
+		{
+			this.IgnoreIfPlatforms(new[]
+			{
+				TestDevice.iOS, TestDevice.Mac, TestDevice.Windows
+			});
+
+			App.WaitForElement("WaitForStubControl");
+
+			// 1. Ensure that the keyboard is closed before we start.
+			App.DismissKeyboard();
+
+			// 2. Scroll to the end.
+			App.WaitForElement(ScrollButton);
+			App.Click(ScrollButton);
+			await Task.Delay(1000); // Wait for the scroll animation to finish.
+			App.ScrollDown("TestScrollView"); // Ensure that we are at the end of the scroll.
+
+			// 3. Focus latest Entry
+			App.WaitForElement(LastEntry);
+			App.EnterText(LastEntry, "test");
+			App.Click(LastEntry);
+			await Task.Delay(1000);
+
+			// 4. The keyboard has opened and the Entry have been translated above the keyboard.
+			App.Screenshot("The keyboard has opened and the Entry have been translated above the keyboard.");
+
+			// 5. Close the keyboard to see if sizes adjust back.
+			App.DismissKeyboard();
+			await Task.Delay(1000);
+
+			// 6. Verify the latest Entry text.
+			var text = App.FindElement(LastEntry).GetText();
+			Assert.AreEqual(text, "test");
+
+			// 7. Make sure that everything has returned to the initial size once the keyboard has closed.
+			App.Screenshot("Make sure that everything has returned to the initial size once the keyboard has closed.");
```

Comment:
How does this validate the scrolling is correct? If you are at the bottom, this will still be visible. Are we able to test bounds to determine the bottom of the entry is near the bottom of the screen - and before it was not near the bottom?

For example, when the keyboard is open, the entry will be say 300px from the bottom, but when the keyboard is closed it will be about 20px from the bottom.

I also don't see the screenshots attached in the test results. Maybe the trailing period if causing issues? If that string is the name, maybe we should also shorten it to say what is happening:

 - 4.: "KeyboardOpenWithEntryAbove"
 - 7.: "KeyBoardClosedWithEntryAtBottom"